File: flutter_app/lib/models/capture_result.dart
import 'dart:typed_data';

class CaptureResult {
  final int status;
  final String path;
  final String windowTitle;
  final Uint8List frameHash;
  final Uint8List prevHash;

  CaptureResult({
    required this.status,
    required this.path,
    required this.windowTitle,
    required this.frameHash,
    required this.prevHash,
  });
}

File: rust_engine/src/types/capture_result.rs
use std::os::raw::c_char;

#[repr(C)]
pub struct CaptureResult {
    pub status: i32,
    pub path: *mut c_char,
    pub window_title: *mut c_char,
    pub frame_hash: [u8; 32],
    pub prev_hash: [u8; 32],
}